app/src/main/java/com/p2p/presentation/tuttifrutti/review/TuttiFruttiReviewRoundAdapter.kt

```diff
@@ -0,0 +1,131 @@
+package com.p2p.presentation.tuttifrutti.review
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewbinding.ViewBinding
+import com.p2p.R
+import com.p2p.databinding.ViewReviewCategoryItemBinding
+import com.p2p.databinding.ViewReviewWordItemBinding
+import com.p2p.model.tuttifrutti.FinishedRoundInfo
+import com.p2p.model.tuttifrutti.FinishedRoundPointsInfo
+import com.p2p.presentation.tuttifrutti.create.categories.Category
+import com.p2p.utils.isEven
+
+/** The adapter used to show the list of round reviews. */
+class TuttiFruttiReviewRoundAdapter(private val onChangeRoundPoints: (String, String, Int) -> Unit) :
+    RecyclerView.Adapter<TuttiFruttiReviewRoundAdapter.RecyclerViewHolder>() {
+
+    var finishedRoundInfo = listOf<FinishedRoundInfo>()
+    var finishedRoundPointsInfo = listOf<FinishedRoundPointsInfo>()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun getItemViewType(position: Int): Int {
+        // If the the modulus number of players plus one is zero, then we get the category title view
+        return position % (finishedRoundInfo.count() + 1)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
+        if (viewType == 0) {
+            return ReviewCategoryViewHolder(
+                ViewReviewCategoryItemBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                )
+            )
+        }
+        return ReviewWordViewHolder(
+            ViewReviewWordItemBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
+        val viewIndex = getItemViewType(position)
+        val viewHolderParams = RecyclerViewHolderParameters()
+        val categoryIndex = getCategoryIndex(position)
+        val viewCategory = getCategory(categoryIndex)
+
+        if (viewIndex == 0) {
+            viewHolderParams.category = viewCategory
+        } else {
+            viewHolderParams.player = finishedRoundInfo[viewIndex - 1].player
+            viewHolderParams.word = finishedRoundInfo[viewIndex - 1].categoriesWords[viewCategory]!!
+            viewHolderParams.points = finishedRoundPointsInfo.find {
+                it.player == viewHolderParams.player
+            }!!.wordsPoints[categoryIndex]
+        }
```

Comment:
Como tip para hacer toda esa `RecyclerViewHolderParameters` inmutable, si querés:

```suggestion
        val viewHolderParams: RecyclerViewHolderParameters
        val categoryIndex = getCategoryIndex(position)
        val viewCategory = getCategory(categoryIndex)

        if (viewIndex == 0) {
            viewHolderParams = RecyclerViewHolderParameters(category = viewCategory)
        } else {
            viewHolderParams = RecyclerViewHolderParameters(
                player = finishedRoundInfo[viewIndex - 1].player,
                word = finishedRoundInfo[viewIndex - 1].categoriesWords.getValue(viewCategory),
                points = finishedRoundPointsInfo.first {
                  it.player == viewHolderParams.player
                }.wordsPoints[categoryIndex]
        }
```

En caso de que prefieras dejarla como está, sí fijate de usar el `getValue` y el `first` en vez de `find` 😅

Reply:
uuuuuuuy genial, muchas gracias!